### ISSUE TITLE
Common preprocessing

### ISSUE THREAD

**Issue by kif** [silx-kit/pyFAI#425]
The preprocessing step is currently done with:

Icor = (I - Idark) / (Flat * Omega * Polarization)

The idea is to have a preprocessing module in charge of doing this, possibly in parallel, shared by all rebinning extension. 

This should take care of dummy values, and support for absorption in the volume of the detector in addition.

**Comment by vallsv** [comment]
And a monitor ?

**Comment by kif** [comment]
yep, in the PR428

duplicate of #398

**Comment by vallsv** [comment]
Done
